### PR TITLE
Histograms to show proportions or percentages

### LINKED ISSUE #5343: Histograms to show proportions or percentages
Currently our histograms show counts (frequencies).  How can we use our dialogues to show proportions or percentages.

There are questions on stack-overflow , for example [here](https://stackoverflow.com/questions/3695497/show-instead-of-counts-in-charts-of-categorical-variables).  

The solution appears to be to add a y to the aes.  I give an example of the additional code below:

It is just the bit in bold that needs to be added, i.e. **y = (..count..)/sum(..count..)**  The rest is easy from the dialogue.  If you multiply by 100 you get percentages and alternatively that can be done with scale_y_continuous(labels = percent).

last_graph <- ggplot2::ggplot(data=mydata, mapping=ggplot2::aes(x=wd1,**y = (..count..)/sum(..count..)**)) + ggplot2::geom_histogram(fill="blue", boundary=-5, bins=36) + theme_grey()+coord_polar(start=0)+ ggplot2::scale_y_continuous(limits=c(-2, 10))

But how can we do this sort of stuff from the dialogue, rather than the script?

### PR DESCRIPTION
**Pull request by Wycklife**
Fixes #5343

@africanmathsinitiative/developers This is ready for review

### CODE REVIEW

**Comment by Carolnikie** [comment]
The dialogue works well in that, when I plot Histogram with  proportions(percentages) it gives the required result. The only problem  I encountered is that if I go back to the same dialogue the percentage check box disappears and it's unchecked as shown in the screenshot below.
s1

**Comment by Wycklife** [comment]
@Carolnikie Thanks for the observation, I'm fixing it.

**Comment by Wycklife** [comment]
@Carolnikie Could you  please check if the issue is fixed?

**Comment by maxwellfundi** [comment]
@Wycklife  @Carolnikie  One thing here that was easy to notice was the size of the plot options button it has become quite big. @Wycklife  I suggest you resize that too. Also I suggest the `Stats` if followed horizontally by the combo box, than the check box for a better design. 

@rdstern  am not completely sure of the results we expect here but they are as follows

**1. Counts**

image

**2. Fractions**

image

**3. fractions  with Percentage checked**

image


**4. Scalled fractions**

image

@Wycklife Sorry I fixed some conflicts with my branch, which have caused conflicts on your branch now, if you could do a new PR from your branch once you have made all the changes, then this can be closed later.

**Comment by maxwellfundi** [comment]
@rdstern  what do you think of these?

**Comment by rdstern** [comment]
Very nice.  Let's merge and try them.

**Comment by maxwellfundi** [comment]
great, @Wycklife please if you could update your master and update your branches, and copy paste your histograms changes, there are some conflicts here.

File: instat/ucrCalculator.vb [hunk 1]
```diff
@@ -893,6 +892,50 @@ Public Class ucrCalculator
...
+            ucrReceiverForCalculation.AddToReceiverAtCursorPosition("lubridate::min()", 1)
+        End If
+    End Sub
+    Private Sub cmdSec_Click(sender As Object, e As EventArgs) Handles cmdSec.Click
+        If chkShowArguments.Checked Then
+            ucrReceiverForCalculation.AddToReceiverAtCursorPosition("lubridate::sec(x= )", 1)
+        Else
+            ucrReceiverForCalculation.AddToReceiverAtCursorPosition("lubridate::sec()", 1)
+        End If
+    End Sub
```
Comment (maxwellfundi) [inline instat/ucrCalculator.vb]: @Wycklife  were these changes meant to be here?

**Comment by maxwellfundi** [comment]
@Wycklife  I would give this priority for this morning, its a small task to do so it shouldn't take so much time. Thanks

**Comment by Wycklife** [comment]
I have opened a new PR with these changes. Am closing this now.

**Pull request closed without merging**